aguaclara/design/floc.py

```diff
@@ -1,307 +1,272 @@
+       specified in many cases. The flocculator will be less than the specified
+       length especially for cases with only one or two sed tanks.
+       channel_L = (vol/(channel_W * downstream_H) + entrancetank_L)/channel_n
+       :returns: Channel length
+       :rtype: float * meter
+       """
+       channel_L = ((self.vol / (self.channel_W * self.downstream_H) + self.entrancetank_L)/self.channel_n).to(u.m)
+       return channel_L
+
+   @property
+   def expansion_max_H(self):
+       """"Return the maximum distance between expansions for the largest
+       allowable H/S ratio.
+       :returns: Maximum expansion distance
+       :rtype: float * meter
+       Examples
+       --------
+       exp_dist_max(20*u.L/u.s, 40*u.cm, 37000, 25*u.degC, 2*u.m)
+       0.375 meter
+       """
+       return (((self.BAFFLE_K / (2 * pc.viscosity_kinematic(self.temp) * (self.vel_grad_avg ** 2))) *
+                (self.Q * self.RATIO_MAX_HS / self.channel_W) ** 3) ** (1/4)).to(u.m)
+
+   @property
+   def expansion_n(self):
+       """Return the minimum number of expansions per baffle space.
+       :returns: Minimum number of expansions/baffle space
+       :rtype: int
+       """
+       return np.ceil(self.downstream_H / self.expansion_max_H)
+
+   @property
+   def expansion_H(self):
+       """Returns the height between flow expansions.
+       :returns: Height between flow expansions
+       :rtype: float * centimeter
+       """
+       return (self.downstream_H / self.expansion_n).to(u.cm)
+
+   @property
+   def baffle_S(self):
+       """Return the spacing between baffles.
+       :returns: Spacing between baffles
+       :rtype: int
+       """
+
+       return ((self.BAFFLE_K /
+                ((2 * self.expansion_H * (self.vel_grad_avg ** 2) *
+                 pc.viscosity_kinematic(self.temp))).to_base_units()) ** (1/3) *
+               self.Q / self.channel_W).to(u.cm)
+
+   @property
+   def obstacle_n(self):
+       """Return the number of obstacles per baffle.
+       :returns: Number of obstacles per baffle
+       :rtype: int
+       """
+       return self.expansion_n - 1
+
+   @property
+   def drain_ND(self):
+       """Returns the diameter of the drain pipe.
+       Each drain pipe will drain two channels because channels are connected by
+       a port at the far end and the first channel can't have a drain because
+       of the entrance tank. Need to review the design to see if this is a good
+       assumption.
+       D_{Pipe} = \sqrt{ \frac{8 A_{Tank}}{\pi t_{Drain}} \sqrt{ \frac{h_0 \sum K}{2g} } }
+       :returns: list of designed values
+       :rtype: float * centimeter
+       """
+
+       drain_K = minorloss.PIPE_ENTRANCE_K_MINOR + minorloss.PIPE_ENTRANCE_K_MINOR +minorloss.PIPE_EXIT_K_MINOR
+       tank_A = 2* self.channel_L * self.channel_W
+       drain_D = (np.sqrt(8 * tank_A/(np.pi* self.drain_t) * np.sqrt(self.downstream_H * drain_K/(2*u.standard_gravity)))).to_base_units()
```

Comment:
A little better, but I think this computation is still a bit too long. Maybe this should actually be using a function defined elsewhere?